packages/toasts/src/ToastsProvider.jsx

```diff
@@ -0,0 +1,39 @@
+import React, { createContext, useState } from 'react';
+import PropTypes from 'prop-types';
+
+export const ToastsContext = createContext();
+
+const ToastsProvider = ({ children }) => {
+  const [toasts, setToasts] = useState([]);
+
+  const addToast = (message) => {
+    setToasts(prevToasts => [
+      ...prevToasts,
+      {
+        id: prevToasts.length,
+        message,
+      },
+    ]);
+  };
+
+  const removeToast = (id) => {
+    const index = toasts.findIndex(toast => toast.id === id);
+    setToasts((prevToasts) => {
+      const newToasts = [...prevToasts];
+      newToasts.splice(index, 1);
+      return newToasts;
+    });
+  };
+
+  return (
+    <ToastsContext.Provider value={{ toasts, addToast, removeToast }}>
```

Comment:
Despite the original implementation not currently being implemented as such, typically context values need to be memoized (with `useMemo`) so that when its parent component re-renders, the object passed into `value` prop doesn't get re-computed each re-render.

Whenever the context value changes, all the children passed to this component re-renders as well. So, the memoization of the context value prevents unintended re-rendering of the children